clang/test/CodeGenSYCL/sycl-framework-metadata2.cpp

```diff
@@ -0,0 +1,27 @@
+// Test checks that !sycl-framework metadata is emitted only to functions
+// whose topmost namespace is sycl.
+
+// RUN: %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown -emit-llvm %s -o - | FileCheck %s
+
+// Check that 'anonymous namespace'::sycl::* functions are not marked with
+// !sycl-framework metadata since topmost namespace is anonymous
+// instead of sycl.
+namespace {
+  namespace sycl {
+    // CHECK-NOT: @_ZN2V14sycl4bar4Ev() {{.*}} !sycl-framework
+    void bar4() {}
+  }
+}
+
+// Check that kernel is not marked with !sycl-framework metadata
+// CHECK-NOT: @_ZTSZ4mainE6kernel() {{.*}} !sycl-framework
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel_single_task(const Func &func) {
```

Comment:
Suggest using the interface from the mock header here too.

Reply:
It leads to error because of ambiguity of sycl namespace and anonymous namespace.